Log is rotated before RPC uniqueness check

### Is it a bug?

- [X] I know this is an issue with the app, and contacting Mullvad support is not relevant.

### I have checked if others have reported this already

- [X] I have checked the issue tracker to see if others have reported similar issues.

### Current Behavior

During the logging system initialization (from `init_daemon_logging`), log file rotation takes place (`rotate_log()`) in `init_logger()` before the RPC uniqueness check takes place (`rpc_uniqueness_check::is_another_instance_running()`). 

I think that in such scenario launching the daemon the second time should cause a failure to init the logging system as it wouldn't be able to rotate the log file from underneath the running daemon. In that sense it would be better to run the RPC check first to establish if another instance is running.

### Expected Behavior

I'd expect the daemon to exit after seeing another instance already running (via IPC uniqueness check)

### Steps to Reproduce

N/A

### Failure Logs

_No response_

### Operating system version

macOS 14.2.1

### Mullvad VPN app version

main branch

### Additional Information

_No response_

To add to that I think the behavior could differ, IIRC: 

- Windows: throws permission error when trying to move a file with open descriptor
- macOS: rotation takes place and current daemon keeps writing into detached file log descriptor.

This is easy to reproduce by running daemon the second time on macOS given that the daemon is already running:

```
sudo /Applications/Mullvad\ VPN.app/Contents/Resources/mullvad-daemon -vv
```

Then all you see in `/var/log/mullvad-vpn/daemon.log` is this:

```
pronebird@codebydre desktop % cat /var/log/mullvad-vpn/daemon.log                                      
[2024-02-05 10:15:25.318][mullvad_daemon::version][INFO] Starting mullvad-daemon - 2023.6 2023-12-06
[2024-02-05 10:15:25.318][mullvad_daemon][INFO] Logging to /var/log/mullvad-vpn
[2024-02-05 10:15:25.318][mullvad_daemon][TRACE] Using configuration: Config { log_level: Trace, log_to_file: true, log_stdout_timestamps: true, launch_daemon_status: false }
[2024-02-05 10:15:25.318][tonic::transport::service::reconnect][TRACE] poll_ready; idle
[2024-02-05 10:15:25.318][tonic::transport::service::reconnect][TRACE] poll_ready; connecting
[2024-02-05 10:15:25.318][tonic::transport::service::reconnect][TRACE] poll_ready; not ready
[2024-02-05 10:15:25.318][tonic::transport::service::reconnect][TRACE] poll_ready; connecting
[2024-02-05 10:15:25.318][tonic::transport::service::reconnect][TRACE] poll_ready; connected
[2024-02-05 10:15:25.318][tonic::transport::service::reconnect][TRACE] poll_ready; not ready
[2024-02-05 10:15:25.318][tonic::transport::service::reconnect][TRACE] poll_ready; connected
[2024-02-05 10:15:25.318][tonic::transport::service::reconnect][TRACE] poll_ready; ready
[2024-02-05 10:15:25.318][mullvad_daemon][ERROR] Another instance of the daemon is already running
[2024-02-05 10:15:25.318][mullvad_daemon][DEBUG] Process exiting with code 1
```

But the actual instance of daemon keeps running and the actual log is seemingly lost.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -38,6 +38,7 @@ Line wrap the file at 100 chars.                                              Th
 ### Fixed
 - Continual excessive attempts to update the API IP were made after testing access methods.
 - Fix pointless API access method rotations for concurrent requests.
+- Fix daemon rotating logs on startup even if another daemon is already running.
 
 ### Security
 #### Android

--- a/mullvad-daemon/src/main.rs
+++ b/mullvad-daemon/src/main.rs
@@ -23,17 +23,23 @@ const EARLY_BOOT_LOG_FILENAME: &str = "early-boot-fw.log";
 
 fn main() {
     let config = cli::get_config();
+
+    let runtime = new_runtime_builder().build().unwrap_or_else(|error| {
+        eprintln!("{}", error.display_chain());
+        std::process::exit(1);
+    });
+
+    if runtime.block_on(rpc_uniqueness_check::is_another_instance_running()) {
+        eprintln!("Another instance of the daemon is already running");
+        std::process::exit(1)
+    }
+
     let log_dir = init_daemon_logging(config).unwrap_or_else(|error| {
         eprintln!("{error}");
         std::process::exit(1)
     });
 
     log::trace!("Using configuration: {:?}", config);
-
-    let runtime = new_runtime_builder().build().unwrap_or_else(|error| {
-        eprintln!("{}", error.display_chain());
-        std::process::exit(1);
-    });
 
     let exit_code = match runtime.block_on(run_platform(config, log_dir)) {
         Ok(_) => 0,
@@ -144,10 +150,6 @@ async fn run_platform(_config: &cli::Config, log_dir: Option<PathBuf>) -> Result
 }
 
 async fn run_standalone(log_dir: Option<PathBuf>) -> Result<(), String> {
-    if rpc_uniqueness_check::is_another_instance_running().await {
-        return Err("Another instance of the daemon is already running".to_owned());
-    }
-
     #[cfg(any(target_os = "macos", target_os = "linux"))]
     if let Err(err) = tokio::fs::remove_file(mullvad_paths::get_rpc_socket_path()).await {
         if err.kind() != std::io::ErrorKind::NotFound {

